compiler/rustc_type_ir/src/lib.rs

```diff
@@ -27,7 +28,7 @@ pub mod ty_info;
 pub mod visit;
 
 #[macro_use]
-mod macros;
+pub mod macros;
```

Comment:
is that `pub` actually needed? :thinking: can `unreachable_phantom_constraint` not be used as long as it itself is `pub`?

Reply:
Will try, but I would have thought not as it will be invoked from (expanded code in) downstream crates.

Reply:
Just to confirm that the module must indeed be `pub` for `unreachable_phantom_constraint` to be reachable from downstream expansions of `prefer_noop_traversal_if_applicable`.